### PR TITLE
Deploy CRC with network isolation

### PR DESCRIPTION
**Pull request by stuggi**
* clone install_yamls
```
git clone https://github.com/openstack-k8s-operators/install_yamls.git
```
* install CRC
```
cd install_yamls/devsetup
CPUS=12 MEMORY=25600 DISK=100 make crc
```
* login to OCP
```
eval $(crc oc-env)
oc login -u kubeadmin -p 12345678 https://api.crc.testing:6443
```
* attach libvirt default network to the crc (default IP 192.168.122.10)
```
make crc_attach_default_interface
```
* install nmstate operator
```
cd ..
make nmstate
```
* configure interfaces on the CRC
```
make nncp
```
* create network-attachment-definitions for internalapi, storage and tenant
```
make netattach
```
the following NADs with ip ranges get configured:
internalapi: 172.17.0.30-172.17.0.70
storage:     172.18.0.30-172.18.0.70
tenant:      172.19.0.30-172.19.0.70

* install metallb operator
```
make metallb
```
* create ipaddresspools (internalapi, storage, tenant) and l2advertisement resources (metallb-sytem namespace)
```
make metallb_config
```
the following IPAddressPools with ip ranges get configured:
internalapi: 172.17.0.80-172.17.0.90
storage:     172.18.0.80-172.18.0.90
tenant:      172.19.0.80-172.19.0.90

* create dependencies
```
make crc_storage
make input
```
* install opentack-operator
```
make openstack
```
* deploy ctlplane using core_v1beta1_openstackcontrolplane_network_isolation.yaml sample, check Depends-On PR 
```
export OPENSTACK_CTLPLANE=config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml make openstack_deploy
```

At this point the ctlplane is deployed with the services using isolated networks as specified in the CR sample.

* create edpm VM (default IP 192.168.122.100)
```
cd devsetup
make edpm_compute
```
* deploy edpm compute
```
make edpm_play
```

* wait until finished, then can check the env
```
oc rsh openstackclient
sh-5.1$ openstack compute service list
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| ID                                   | Binary         | Host                   | Zone     | Status  | State | Updated At                 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| 25ce958d-5c7b-47b8-a98a-8b84fb7e8790 | nova-conductor | nova-cell0-conductor-0 | internal | enabled | up    | 2023-03-06T15:29:09.000000 |
| 0ccc4465-f382-44db-8130-82503ad49a9a | nova-scheduler | nova-scheduler-0       | internal | enabled | up    | 2023-03-06T15:29:09.000000 |
| b64d6016-9c58-4657-8a4d-6fc5d73a4908 | nova-conductor | nova-cell1-conductor-0 | internal | enabled | up    | 2023-03-06T15:29:06.000000 |
| db60bfe7-be1a-4b8b-9939-5faf368ba96e | nova-compute   | edpm-compute-0         | nova     | enabled | up    | 2023-03-06T15:29:08.000000 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
```

Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/215

### CODE REVIEW

File: scripts/gen-edpm.sh [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    deploy: true
+    ansibleVars: |
+      edpm_network_config_template: templates/single_nic_vlans/single_nic_vlans.j2
+      edpm_network_config_hide_sensitive_logs: false
+      edpm_ovn_dbs:
+      - ${EDPM_OVN_DBS}
+      edpm_hosts_entries_extra_hosts_entries:
+      - 172.17.0.80 glance-internal.openstack.svc neutron-internal.openstack.svc cinder-internal.openstack.svc nova-internal.openstack.svc placement-internal.openstack.svc keystone-internal.openstack.svc
+      - 172.17.0.85 rabbitmq.openstack.svc
+      - 172.17.0.86 rabbitmq-cell1.openstack.svc
```
Comment (stuggi) [inline scripts/gen-edpm.sh:113]: the role is not triggered yet

File: devsetup/Makefile [hunk 2]
```diff
@@ -44,11 +45,20 @@ download_tools: ## Runs an ansible playbook to install required tools with the v
 	download_tools.yaml
 
 .PHONY: crc_attach_default_interface
-crc_attach_default_interface:
-	sudo virsh attach-interface crc --source default --type network --model virtio --config --persistent
+crc_attach_default_interface: ## Attach default libvirt network to CRC
+	-CURRENT_MAC=$(shell sudo virsh net-dumpxml default | grep crc | sed -e "s/.*mac='\(.*\)' name.*/\1/"); \
```
Comment (gibizer) [inline devsetup/Makefile:49]: I don't see how this finds `crc` in the default network definition.
```
gibi@bedrock:~                                                                                                                                                                                      $ sudo virsh net-dumpxml default
<network>
  <name>default</name>
  <uuid>d8d921ab-df9b-4626-b63c-cfdba21ffb07</uuid>
  <forward mode='nat'/>
  <bridge name='virbr0' stp='on' delay='0'/>
  <mac address='52:54:00:a8:34:cb'/>
  <ip address='192.168.122.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.122.2' end='192.168.122.254'/>
    </dhcp>
  </ip>
</network>

gibi@bedrock:~                                                                                                                                                                                      $ sudo virsh net-dumpxml crc
<network connections='1'>
  <name>crc</name>
  <uuid>49eee855-d342-46c3-9ed3-b8d1758814cd</uuid>
  <forward mode='nat'>
    <nat>
      <port start='1024' end='65535'/>
    </nat>
  </forward>
  <bridge name='crc' stp='on' delay='0'/>
  <mac address='52:54:00:fd:be:d0'/>
  <ip family='ipv4' address='192.168.130.1' prefix='24'>
    <dhcp>
      <host mac='52:fd:fc:07:21:82' ip='192.168.130.11'/>
    </dhcp>
  </ip>
</network>

```

Comment (stuggi) [reply devsetup/Makefile:49]: is this output from after running this step? 
Getting the CURRENT_MAC is to do cleanup when rerunning this step to
* first detach and remove the entry from the network and don't fail if it does not exist using the '-' prefix
* second to attach a new interface with a new MAC address

This is to allow if someone reruns with a different CRC_DEFAULT_NETWORK_IP.

File: Makefile [hunk 3]
```diff
@@ -1036,3 +1039,88 @@ ceph_cleanup: ## deletes the ceph pod
...
+	$(eval $(call vars,$@,edpm_deploy))
+	bash scripts/gen-edpm.sh
+	oc apply -f ${DEPLOY_DIR}/
+	# TODO
+	# oc wait openstackdataplanenode -n ${NAMESPACE} -l component=openstackdataplanenode --for condition=Ready --timeout=600s
+
+.PHONY: edpm_deploy_cleanup
+edpm_deploy_cleanup: ## Deletes the OpenStackDataPlaneNode resource, does not delete the edpm node
+	$(eval $(call vars,$@,edpm_deploy))
+	oc delete --ignore-not-found=true -f ${DEPLOY_DIR}/
```
Comment (stuggi) [inline Makefile:1125]: not completely sure if edpm deployment under devsetup dir as it is now with https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/Makefile#L60

File: devsetup/scripts/gen-edpm-compute-node.sh [hunk 4]
```diff
@@ -148,6 +148,16 @@ cat <<EOF >../out/edpm/${EDPM_COMPUTE_NAME}.xml
 </domain>
 EOF
 
+cat <<EOF >../out/edpm/tripleocurrent.repo
```
Comment (stuggi) [inline devsetup/scripts/gen-edpm-compute-node.sh:151]: oh just have seen that this is currently done in https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/edpm/edpm-play.yaml#L14

File: Makefile (hunk 3 above)
Comment (stuggi) [reply Makefile:1125]: I might remove this and test integrate with current `make edpm_play` from devsetup

File: devsetup/Makefile (hunk 2 above)
Comment (stuggi) [reply devsetup/Makefile:49]: I'll split this target to have a `crc_attach_default_interface_cleanup`. this aligns better with what we have in all the other make targets

Comment (gibizer) [reply devsetup/Makefile:49]: I guess I misunderstood the error messages coming from this make target. I try to follow the procedure you describe in the PR message and I see this locally after `crc start` finished successfully.

```console
$ make crc_attach_default_interface
CURRENT_MAC=; \
sudo virsh detach-interface crc network --mac "$CURRENT_MAC"
error: Failed to get option 'mac': Option argument is empty
error: Failed to detach interface

make: [Makefile:49: crc_attach_default_interface] Error 1 (ignored)
sudo virsh net-update default delete ip-dhcp-host "<host name='crc'/>" --config --live
error: Failed to update network default
error: Requested operation is not valid: network is not running

make: [Makefile:50: crc_attach_default_interface] Error 1 (ignored)
sleep 5
MAC_ADDRESS=52:54:00:AD:A8:F5; \
sudo virsh net-update default add-last ip-dhcp-host --xml "<host mac='$MAC_ADDRESS' name='crc' ip='192.168.122.10'/>" --config --live; \
sudo virsh attach-interface crc --source default --type network --model virtio --mac $MAC_ADDRESS --config --persistent; \
sleep 10; \
WORKER=crc-rwwzd-master-0; \
oc debug node/$WORKER -- ip -o link | awk 'toupper($0) ~ /ETHER $MAC_ADDRESS/{print $2}'  | awk -F: '{print $1}'
error: Failed to update network default
error: Requested operation is not valid: network is not running

error: Failed to attach interface
error: Requested operation is not valid: network 'default' is not active

Temporary namespace openshift-debug-9v68k is created for debugging node...
Starting pod/crc-rwwzd-master-0-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
Temporary namespace openshift-debug-9v68k was removed.
```

Maybe it is more about the status of the `default` network on my machine

Comment (gibizer) [reply devsetup/Makefile:49]:
```
$ sudo virsh net-list --all
 Name      State      Autostart   Persistent
----------------------------------------------
 crc       active     yes         yes
 default   inactive   no          yes

```

Comment (gibizer) [reply devsetup/Makefile:49]:
``` console
$ sudo virsh net-start default
Network default started
```

```console
$ make crc_attach_default_interface
CURRENT_MAC=; \
sudo virsh detach-interface crc network --mac "$CURRENT_MAC"
error: Failed to get option 'mac': Option argument is empty
error: Failed to detach interface

make: [Makefile:49: crc_attach_default_interface] Error 1 (ignored)
sudo virsh net-update default delete ip-dhcp-host "<host name='crc'/>" --config --live
error: Failed to update network default
error: Requested operation is not valid: couldn't locate a matching dhcp host entry in network 'default'

make: [Makefile:50: crc_attach_default_interface] Error 1 (ignored)
sleep 5
MAC_ADDRESS=52:54:00:8B:3B:5A; \
sudo virsh net-update default add-last ip-dhcp-host --xml "<host mac='$MAC_ADDRESS' name='crc' ip='192.168.122.10'/>" --config --live; \
sudo virsh attach-interface crc --source default --type network --model virtio --mac $MAC_ADDRESS --config --persistent; \
sleep 10; \
WORKER=crc-rwwzd-master-0; \
oc debug node/$WORKER -- ip -o link | awk 'toupper($0) ~ /ETHER $MAC_ADDRESS/{print $2}'  | awk -F: '{print $1}'
Updated network default persistent config and live state
Interface attached successfully

Temporary namespace openshift-debug-lrr6s is created for debugging node...
Starting pod/crc-rwwzd-master-0-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
Temporary namespace openshift-debug-lrr6s was removed.
```

Comment (gibizer) [reply devsetup/Makefile:49]: OK, it seems at least the interface get added to the crc domain and got an ip:
```
$ sudo virsh domifaddr crc
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet0      52:fd:fc:07:21:82    ipv4         192.168.130.11/24
 vnet1      52:54:00:8b:3b:5a    ipv4         192.168.122.10/24
```

**Comment by stuggi** [comment]
I have updated the PR to
* add cleanup targets for the networking resources
* use the existing `make edpm_play for now instead of using the dataplay operator node deploy resource

With this the compute node successfully registers to the ctlplane. The network agents don't, so need to be checked
```
oc rsh openstackclient
sh-5.1$ openstack compute service list
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| ID                                   | Binary         | Host                   | Zone     | Status  | State | Updated At                 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| 25ce958d-5c7b-47b8-a98a-8b84fb7e8790 | nova-conductor | nova-cell0-conductor-0 | internal | enabled | up    | 2023-03-06T15:29:09.000000 |
| 0ccc4465-f382-44db-8130-82503ad49a9a | nova-scheduler | nova-scheduler-0       | internal | enabled | up    | 2023-03-06T15:29:09.000000 |
| b64d6016-9c58-4657-8a4d-6fc5d73a4908 | nova-conductor | nova-cell1-conductor-0 | internal | enabled | up    | 2023-03-06T15:29:06.000000 |
| db60bfe7-be1a-4b8b-9939-5faf368ba96e | nova-compute   | edpm-compute-0         | nova     | enabled | up    | 2023-03-06T15:29:08.000000 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
```

File: Makefile [hunk 5]
```diff
@@ -1036,3 +1039,112 @@ ceph_cleanup: ## deletes the ceph pod
...
+metallb_config: export INTERFACE=${NNCP_INTERFACE}
+metallb_config: ## creates the IPAddressPools and l2advertisement resources
+	$(eval $(call vars,$@,metallb))
+	bash scripts/gen-olm-metallb.sh
+	oc apply -f ${DEPLOY_DIR}/ipaddresspools.yaml
+	oc apply -f ${DEPLOY_DIR}/l2advertisement.yaml
+
+.PHONY: metallb_config_cleanup
+metallb_config_cleanup: export NAMESPACE=metallb-system
+metallb_config_cleanup: ## creates the IPAddressPools and l2advertisement resources
```
Comment (abays) [inline Makefile:1123]: Comment mismatch

File: Makefile [hunk 6]
```diff
@@ -1036,3 +1039,112 @@ ceph_cleanup: ## deletes the ceph pod
...
+	sleep 2
+	bash scripts/gen-olm-metallb.sh
+	oc apply -f ${OPERATOR_DIR}
+	while ! (oc get pod --no-headers=true -l control-plane=controller-manager -n ${NAMESPACE}| grep "metallb-operator-controller"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} --for condition=Ready -l control-plane=controller-manager --timeout=180s
+	while ! (oc get pod --no-headers=true -l component=webhook-server -n ${NAMESPACE}| grep "metallb-operator-webhook"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} --for condition=Ready -l component=webhook-server --timeout=180s
+	oc apply -f ${DEPLOY_DIR}/deploy_operator.yaml
+	while ! (oc get pod --no-headers=true -l component=speaker -n ${NAMESPACE} | grep "speaker"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} -l component=speaker --for condition=Ready --timeout=180s
```
Comment (dprince) [inline Makefile:1110]: I think it is probably fine to have these here for development. But I wonder if we could get by with just adding a requirement on the CSV like this: https://github.com/dprince/openstack-operator/commit/5c30f31d5d2e5df8c65fed58a0c1a32ef585e99a I think we'll want/need to do something like this eventually for the product as well right?

File: Makefile [hunk 7]
```diff
@@ -1036,3 +1039,112 @@ ceph_cleanup: ## deletes the ceph pod
...
+
+.PHONY: netattach
+netattach: export INTERFACE=${NNCP_INTERFACE}
+netattach: namespace ## Creates network-attachment-definitions for the networks the workers are attached via nncp
+	$(eval $(call vars,$@,netattach))
+	bash scripts/gen-netatt.sh
+	oc apply -f ${DEPLOY_DIR}/
+
+.PHONY: netattach_cleanup
+netattach_cleanup: ## Creates network-attachment-definitions for the networks the workers are attached via nncp
```
[the remark this answers is not in the document]
Comment (stuggi) [reply Makefile]: done

File: Makefile (hunk 5 above)
Comment (stuggi) [reply Makefile]: done

File: scripts/gen-olm-nmstate.sh [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+echo DEPLOY_DIR ${DEPLOY_DIR}
+
+cat > ${OPERATOR_DIR}/operatorgroup.yaml <<EOF_CAT
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  annotations:
+    olm.providedAPIs: NMState.v1.nmstate.io
+  generateName: openshift-nmstate-
+  name: openshift-nmstate-tn6k8
```
Comment (gibizer) [inline scripts/gen-olm-nmstate.sh:44]: `tn6k8` seems to be a some random generated name. I'm wondering if we need that in the name here.

**Review by gibizer — COMMENTED** [review]
Overall this looks good to me. I was able to deploy a control plane based on the steps in the PR description. I'm working through the edpm deploy part.

**Review by gibizer — COMMENTED** [review]
OK, I was able to deploy the edpm side of it too. 

One note that is not related to this patch but I noticed that nova-compute is deployed with `reserved_host_memory_mb = 4096`  in a VM that has total 4G of ram. So while nova-compute starts it fails to create the compute RP in placement as nova tries to reserve 4G out of the  3662MB it sees.

File: scripts/gen-olm-nmstate.sh (hunk 8 above)
Comment (stuggi) [reply scripts/gen-olm-nmstate.sh:44]: I just took the one from the example here https://docs.openshift.com/container-platform/4.12/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html

**Comment by stuggi** [comment]
there is a bug in the ovs controller `OPENSTACK_IMG=quay.io/mschuppe/openstack-operator-index:a8a8eb0c567344f97da198fabf3558dd50c564a1` installs the operator which has the current proposed fix so that ovn controller agents registration ok.

> OK, I was able to deploy the edpm side of it too.
> 
> One note that is not related to this patch but I noticed that nova-compute is deployed with `reserved_host_memory_mb = 4096` in a VM that has total 4G of ram. So while nova-compute starts it fails to create the compute RP in placement as nova tries to reserve 4G out of the 3662MB it sees.

good catch, the default is `edpm_nova_compute_reserved_host_memory: 4096`. I'll update the PR to set it to 2048 for now.

File: Makefile (hunk 6 above)
Comment (stuggi) [reply Makefile:1110]: yes I think we want to do that

**Comment by stuggi** [comment]
Test creating instance and access it

```
oc rsh openstackclient
openstack network create private --share
openstack subnet create priv_sub --subnet-range 192.168.0.0/24 --network private
openstack flavor create --ram 512 --vcpus 1 --disk 1 --ephemeral 1 m1.small
curl -L -o /tmp/cirros-0.5.2-x86_64-disk.img http://download.cirros-cloud.net/0.5.2/cirros-0.5.2-x86_64-disk.img
openstack image create --container-format bare --disk-format qcow2 --file /tmp/cirros-0.5.2-x86_64-disk.img cirros
openstack server create --flavor m1.small --image cirros --nic net-id=$(openstack network list --name private -f value -c ID) test
openstack network create public --external  --provider-network-type flat --provider-physical-network datacentre
openstack subnet create pub_sub --subnet-range 192.168.122.0/24 --allocation-pool start=192.168.122.200,end=192.168.122.210 --gateway 192.168.122.1 --no-dhcp --network public
openstack router create priv_router
openstack router add subnet priv_router priv_sub
openstack router set priv_router --external-gateway public
openstack floating ip create public
=>  192.168.122.207
openstack server add floating ip test 192.168.122.207

openstack server list
+--------------------------------------+------+--------+---------------------------------------+--------+----------+
| ID                                   | Name | Status | Networks                              | Image  | Flavor   |
+--------------------------------------+------+--------+---------------------------------------+--------+----------+
| a45e1674-cc72-4b19-b852-2d0d44f2e9c9 | test | ACTIVE | private=192.168.0.77, 192.168.122.207 | cirros | m1.small |
+--------------------------------------+------+--------+---------------------------------------+--------+----------+

openstack security group rule create --protocol icmp --ingress --icmp-type -1 default
openstack security group rule create --protocol tcp --ingress --dst-port 22

sh-5.1$ ping 192.168.122.207
PING 192.168.122.207 (192.168.122.207) 56(84) bytes of data.
64 bytes from 192.168.122.207: icmp_seq=1 ttl=62 time=4.70 ms
64 bytes from 192.168.122.207: icmp_seq=2 ttl=62 time=1.40 ms

sh-5.1$ ssh cirros@192.168.122.207
The authenticity of host '192.168.122.207 (192.168.122.207)' can't be established.
ECDSA key fingerprint is SHA256:+U92Nj51UMzfcQ0l8XtglL74D5CIpOzC8rtMpe6L4ok.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added '192.168.122.207' (ECDSA) to the list of known hosts.
cirros@192.168.122.207's password: 
$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1442 qdisc pfifo_fast qlen 1000
    link/ether fa:16:3e:c0:9f:8c brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.77/24 brd 192.168.0.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::f816:3eff:fec0:9f8c/64 scope link 
       valid_lft forever preferred_lft forever
```

File: Makefile [hunk 9]
```diff
@@ -1036,3 +1039,89 @@ ceph_cleanup: ## deletes the ceph pod
...
+nmstate: export NAMESPACE=openshift-nmstate
+nmstate: ## installs nmstate operator in the openshift-nmstate namespace
+	$(eval $(call vars,$@,nmstate))
+	bash scripts/gen-namespace.sh
+	oc apply -f ${OUT}/${NAMESPACE}/namespace.yaml
+	sleep 2
+	bash scripts/gen-olm-nmstate.sh
+	oc apply -f ${OPERATOR_DIR}
+	while ! (oc get pod --no-headers=true -l app=kubernetes-nmstate-operator -n ${NAMESPACE}| grep "nmstate-operator"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} --for condition=Ready -l app=kubernetes-nmstate-operator --timeout=180s
```
Comment (olliewalsh) [inline Makefile:1054]: timeout was too short for my local deployment

Comment (stuggi) [reply Makefile]: ok, I bumped them to be 300s, as its a max it does not hurt

File: devsetup/edpm/edpm-play.yaml [hunk 10]
```diff
@@ -172,10 +204,10 @@ spec:
                   edpm_nova_compute_service_user_username: nova
                   edpm_nova_compute_vnc_novncproxy_base_url: http://192.168.24.3:6080
                   edpm_ovn_dbs:
-                  - 192.168.24.1
+                  - _EDPM_OVN_DBS_
             vars:
                 registry_name: "quay.io"
-                registry_namespace: "tripleomastercentos9"
+                registry_namespace: "tripleozedcentos9"
```
Comment (bogdando) [inline devsetup/edpm/edpm-play.yaml:210]: I think it should be based on Wallaby as tripleo master/zed containers are going to become unsupported shortly? @rlandy WDYT?

Comment (stuggi) [reply devsetup/edpm/edpm-play.yaml:210]: our current work is based on zed. and we default to those also in the operators

Comment (bogdando) [reply devsetup/edpm/edpm-play.yaml:210]: I confused this to the Wallaby base used for adoption, sorry

File: devsetup/Makefile [hunk 11]
```diff
@@ -36,35 +45,58 @@ crc_cleanup: ## Destroys the CRC env, but does NOT clear ( --clear-cache ) the c
...
+	sleep 10; \
+	WORKER=$(shell oc get nodes -l node-role.kubernetes.io/worker -o jsonpath="{.items[*].metadata.name}"); \
+	oc debug node/$$WORKER -- ip -o link | awk 'toupper($$0) ~ /ETHER $$MAC_ADDRESS/{print $$2}'  | awk -F: '{print $$1}'
+
+.PHONY: crc_attach_default_interface_cleanup
+crc_attach_default_interface_cleanup: ## Detach default libvirt network from CRC
+	-MAC_ADDRESS=$(shell sudo virsh net-dumpxml default | grep crc | sed -e "s/.*mac='\(.*\)' name.*/\1/"); \
+	sudo virsh detach-interface crc network --mac "$$MAC_ADDRESS"
+	-sudo virsh net-update default delete ip-dhcp-host "<host name='crc'/>" --config --live
+	sleep 5
```
[the remark this answers is not in the document]
Comment (stuggi) [reply devsetup/Makefile:62]: I don't know the vagrant/director_standalone setup, but I think in general you want to use the on both sides the same libvirt network and same VLANs/networks. right now it is the default libvirt network which is used for CRC and edpm nodes for that. on your vagrant vm is it the director_standalone0 network which is used to create the vlans on? I don't know the vagrant setup, but I think you'd either need to use crc/edpm vms to use the director-standalone0 network or the vagrant vm the default network.

**Pull request merged**